File: backend/schemas/incident.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class IncidentBase(BaseModel):
    type: str
    emoji: str
    title: str
    summary: str | None = None
    source_type: str
    confidence_tier: str
    confidence_score: float
    latitude: float
    longitude: float
    grid_cell: str
    timestamp_first_seen: datetime
    timestamp_last_updated: datetime
    number_of_reports: int
    official_overlap: bool
    media_count: int
    tags: list[str] = []
    verification_notes: str | None = None
    is_active: bool


class IncidentResponse(IncidentBase):
    id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class IncidentListResponse(BaseModel):
    total: int
    items: list[IncidentResponse]


class IncidentMapResponse(BaseModel):
    id: str
    type: str
    emoji: str
    title: str
    confidence_tier: str
    confidence_score: float
    latitude: float
    longitude: float
    number_of_reports: int
    is_active: bool

    model_config = ConfigDict(from_attributes=True)

File: web/types/incident.ts
export interface Incident {
  id: string;
  type: string;
  title: string;
  summary: string;
  source_type: string;
  confidence_tier: "official" | "corroborated" | "plausible" | "unverified";
  confidence_score: number;
  latitude: number;
  longitude: number;
  grid_cell: string;
  timestamp_first_seen: string;
  timestamp_last_updated: string;
  number_of_reports: number;
  official_overlap: boolean;
  media_count: number;
  tags: string[];
  verification_notes: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface IncidentListResponse {
  total: number;
  items: Incident[];
}

export interface MapIncident {
  id: string;
  type: string;
  title: string;
  confidence_tier: "official" | "corroborated" | "plausible" | "unverified";
  confidence_score: number;
  latitude: number;
  longitude: number;
  number_of_reports: number;
  is_active: boolean;
}
